src/scenes/App/components/TermsAndConditions/index.js

```diff
@@ -0,0 +1,265 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+import { connect } from 'react-redux';
+import { Paper, Typography, Grid, FormGroup, FormControlLabel, Checkbox, Button, withStyles } from 'material-ui';
+import { FormattedMessage, injectIntl, intlShape } from 'react-intl';
+
+import AppConfig from '../../../../config/app';
+import styles from './styles';
+
+const TNC_ACCEPTED = 'tncAccepted';
+
+@injectIntl
+@withStyles(styles, { withTheme: true })
+export default class TermsAndConditions extends Component {
+  static propTypes = {
+    classes: PropTypes.object.isRequired,
+    intl: intlShape.isRequired, // eslint-disable-line react/no-typos
+    langHandler: PropTypes.func,
+  };
+
+  static defaultProps = {
+    langHandler: undefined,
+  };
+
+  state = {
+    isVisible: !!localStorage.getItem(TNC_ACCEPTED),
+    checkboxChecked: false,
+  };
+
+  render() {
+    const { classes, intl: { locale } } = this.props;
+    const { isVisible, checkboxChecked } = this.state;
+
+    return (
+      <div
+        className={classes.tncBg}
+        style={{
+          opacity: isVisible ? 0 : 1,
+          display: isVisible ? 'none' : 'block',
+        }}
+      >
+        <Grid container>
+          <Grid item xs>
+            <Button color="primary" variant="raised" className={classes.langBtn} onClick={this.props.langHandler}>
+              <FormattedMessage id="language.select" defaultMessage="中文" />
+            </Button>
+          </Grid>
+          <Grid item xs={8}>
+            <Paper className={classes.tncWrapper}>
+              <Heading />
+              <GeneralSection />
+              <DefinitionsSection />
+              <SubordinationSection />
+              <ParticipateSection />
+              <RepresentationsSection />
+              <AwardsSection />
+              <PowersSection />
+              <ResponsibilitiesSection />
+              <IpSection />
+              <SupportSection />
+              <MiscSection />
+            </Paper>
+            <FormGroup row className={classes.formGroup}>
+              <FormControlLabel
+                control={
+                  <Checkbox
+                    checked={checkboxChecked}
+                    value="checkboxChecked"
+                    onChange={this.handleCheckedChange('checkboxChecked')}
```

Comment:
do we need this method? Can't we just do `onChange={(e) => this.setState({ checkboxChecked: e.target.value })}`? Sorry if you commented on this already and I didn't see it

Reply:
I prefer to keep this logic non-inlined. Trying to maintain the single responsibility principle.